src/rust/engine/src/externs/interface.rs

```diff
@@ -0,0 +1,1608 @@
+// Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+// Licensed under the Apache License, Version 2.0 (see LICENSE).
+
```

Comment:
This is not generated, right? It's pretty big so didn't do a close review. Looks good from a skim. Lmk if I should ready closely.

Reply:
This is the file formerly known as `engine_cffi/src/lib.rs`: if you look at the commits independently, you can mostly the second one.